### PR TITLE
[Model] Skip unused Jina V5 output layers

### PR DESCRIPTION
**Pull request by BabyDrangoner**
## Purpose

Jina Embeddings V5 models are pooling-only, but their vLLM wrappers inherit
causal-LM classes. Because these wrappers already declare themselves as pooling
models, they bypass the generic pooling adapter that replaces generation-only
output layers. The encoder/nano variant therefore retained an unused
`ParallelLMHead` with shape `[128256, 768]`.

This change applies the existing pooling-model `no_init_weights` pattern to
both Jina V5 wrappers, replacing `ParallelLMHead` and `LogitsProcessor` with a
`StageMissingLayer`. The decoder/small checkpoint ties its output head to the
input embedding (`VocabParallelEmbedding`), so that shared embedding remains
intact.

On an NVIDIA RTX PRO 6000 Blackwell (TP=1), the nano checkpoint's loaded model
state changed as follows:

| Metric | Before | After |
|---|---:|---:|
| Registered parameter bytes | 620,533,248 | 423,532,032 |
| Unused `lm_head` bytes | 197,001,216 | 0 |
| CUDA memory delta after model load | 1,283,457,024 | 1,086,324,736 |

This removes 187.875 MiB of resident parameters (31.75%) and approximately
188 MiB of post-load CUDA memory. This is a resident-memory result; it is not a
claim that the transient module-construction peak is eliminated.

I searched open vLLM PRs and issues for Jina V5/Embeddings V5 combined with
`lm_head`, output-layer, and pooling terms and found no direct or semantic
duplicate. #32757 is the already-merged generic pooling mechanism used as the
implementation precedent, not another Jina-specific fix.

AI assistance (OpenAI Codex) was used to help investigate, implement, and test
this change. I reviewed the complete diff and the validation results.

## Test Plan

- Load and embed two natural prompts with both the base and patched
  `jinaai/jina-embeddings-v5-text-nano` checkpoints; compare parameter and
  post-load CUDA memory plus float32 embedding bytes.
- Repeat the embedding comparison with
  `jinaai/jina-embeddings-v5-text-small` to cover its tied embedding.
- Run the existing Jina pooling-model, adapter, and registry tests, then all
  pre-commit hooks applicable to the changed production file.

Commands:

```bash
.venv/bin/python -m pytest -q \
  tests/models/language/pooling/test_jina_embeddings_v5.py \
  tests/models/test_adapters.py \
  'tests/models/test_registry.py::test_registry_imports[JinaEmbeddingsV5Model]'

uvx pre-commit run --files \
  vllm/model_executor/models/jina.py

git diff --check
```

## Test Result

- Pytest: **12 passed**.
- Pre-commit: all applicable hooks passed, including Ruff, formatting, mypy,
  SPDX, forbidden-import, and configuration checks.
- Nano real-model A/B:
  - Both runs produced embedding SHA-256
    `f355f21e03004acfc86af056b3610282290563ab563dd5f9d04e5768bca06688`.
  - Both output vectors had shape `[768]` and L2 norm `1.0`.
  - Resident parameter and CUDA memory reductions are reported above.
- Small real-model A/B:
  - Both runs retained 1,192,099,840 parameter bytes and no independent
    `lm_head`, confirming the tied input embedding was not removed.
  - Both produced embedding SHA-256
    `c740490981d58e61cee407638b98c0ad724674e11341bc94086a7fa19a675942`.
  - Both output vectors had shape `[1024]` and L2 norm `1.0`.

### CODE REVIEW

File: tests/models/language/pooling/test_jina_embeddings_v5.py [hunk 1]
```diff
@@ -65,3 +73,40 @@ def test_supported_decoder_backbone_is_accepted():
...
+        (JinaEmbeddingsV5EncoderModel, LlamaForCausalLM),
+    ],
+)
+def test_pooling_model_skips_output_layer(monkeypatch, model_cls, base_cls):
+    class FakeLMHead(torch.nn.Linear):
+        pass
+
+    class FakeLogitsProcessor(torch.nn.Module):
+        pass
+
```
Comment (noooop) [inline tests/models/language/pooling/test_jina_embeddings_v5.py]: Thanks for your contribution.

I don't think we need to do regression testing on this.

Comment (BabyDrangoner) [reply tests/models/language/pooling/test_jina_embeddings_v5.py]: Agreed. I removed the dedicated regression test and its imports in 2f5766a43, leaving the PR with only the production change. The existing related suites still pass (12 passed), all applicable pre-commit hooks pass, and I updated the PR description accordingly.

**Review by noooop — APPROVED** [review]
Thanks for your contribution.
